frontend/src/HomePage/AppMenu.jsx

```diff
@@ -40,9 +42,11 @@ export const AppMenu = function AppMenu({
   return (
     <OverlayTrigger
       trigger="click"
-      placement="bottom-end"
+      placement="top-start"
       rootClose
       onToggle={onMenuOpen}
+      onExit={() => setMenuOpen(false)}
+      show={popoverVisible}
```

Comment:
```suggestion
      container={document.getElementsByClassName('home-page-content')[0]}
```

Adding the container might fix the issue

Reply:
should I set the container placement to `top start` as default or calculate if the container is going out of view and then only set the placement to `top-start`

Reply:
Adding the container did fix the issue but only keeping the `container` logic leads to the app menu following the scroll and the menu does not close automatically.
I have attached both views in the below recording, if you want i can either keep the previous changes or revert it back.

https://github.com/user-attachments/assets/199831bd-1388-42fc-8fc0-87d87642b2bd